src/saltext/vmware/states/vmc_distributed_firewall_rules.py

```diff
@@ -0,0 +1,487 @@
+                name=name,
+                comment="Updated distributed firewall rule {}".format(rule_id),
+                old_state=distributed_firewall_rule,
+                new_state=updated_distributed_firewall_rule,
+                result=True,
+            )
+        else:
+            log.info("All fields are same as existing distributed firewall rule %s", rule_id)
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Distributed firewall rule exists already, no action to perform",
+                result=True,
+            )
+    else:
+        log.info("No distributed firewall rule found with ID %s", rule_id)
+        created_distributed_firewall_rule = __salt__["vmc_distributed_firewall_rules.create"](
+            hostname=hostname,
+            refresh_key=refresh_key,
+            authorization_host=authorization_host,
+            org_id=org_id,
+            sddc_id=sddc_id,
+            domain_id=domain_id,
+            security_policy_id=security_policy_id,
+            rule_id=rule_id,
+            verify_ssl=verify_ssl,
+            cert=cert,
+            source_groups=source_groups,
+            destination_groups=destination_groups,
+            services=services,
+            scope=scope,
+            action=action,
+            sequence_number=sequence_number,
+            disabled=disabled,
+            logged=logged,
+            description=description,
+            direction=direction,
+            notes=notes,
+            tag=tag,
+            tags=tags,
+        )
+
+        if "error" in created_distributed_firewall_rule:
+            return vmc_state._create_state_response(
+                name=name, comment=created_distributed_firewall_rule["error"], result=False
+            )
+
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Created distributed firewall rule {}".format(rule_id),
+            new_state=created_distributed_firewall_rule,
+            result=True,
+        )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    domain_id,
+    security_policy_id,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given distributed firewall rule does not exist on given SDDC
+
+    name
+        Indicates the distributed firewall rule id, any unique string identifying the distributed firewall rule.
+
+    hostname
+        The host name of NSX-T manager
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console
+
+    org_id
+        The Id of organization to which the SDDC belongs to
+
+    sddc_id
+        The Id of SDDC from which the distributed firewall rule should be deleted
+
+    domain_id
+        The domain_id for which the Distributed firewall rule belongs to. Possible values: default, cgw
+
```

Comment:
**question** where is the security_policy_id docs?

Reply:
missed to add earlier. now updated it